crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3391,14 +3405,33 @@ impl<'db> TypeInferenceBuilder<'db> {
     fn infer_attribute_expression(&mut self, attribute: &ast::ExprAttribute) -> Type<'db> {
         let ast::ExprAttribute {
             value,
-            attr: _,
+            attr,
             range: _,
             ctx,
         } = attribute;
 
         match ctx {
             ExprContext::Load => self.infer_attribute_load(attribute),
-            ExprContext::Store | ExprContext::Del => {
+            ExprContext::Store => {
+                let value_ty = self.infer_expression(value);
+
+                if let (ast::ExprContext::Store, Type::Instance(instance)) = (ctx, value_ty) {
```

Comment:
We're already inside a match arm for `ctx` being `Store`, why do we need to match on `ctx` again?

Reply:
Thanks for catching that. I previously had this code block in an entirely different place (where the match on `ctx` was necessary) and forgot to remove it when I moved it here.